man/systemd.link.xml

```diff
@@ -409,6 +409,14 @@
           be ignored.</para>
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><varname>TxQueueLen=</varname></term>
+        <listitem>
+          <para>Specifies the transmit queue length of the device in number of packets. An unsigned integer
+          in the range 0..4294967295. When unset (equivalent to explicitely setting it to 0), the kernel's
+          default will be used.</para>
```

Comment:
don't make "0" magic here. 

Our usual way to reset values to their defaults is to assign an empty string. do that here too. i.e. refuse "0", but set the internal value to 0 if the empty string is assigned.

i.e. the fact that 0 means "kernel default" should be an internal detail, and not exposed to the user. and we should stick to the same logic everywhere for resetting values to their defaults: empty string.

Reply:
Thanks for clarifying that to me: I studied diverse related pull requests afterwards and saw that you did not explain this for the first time... unfortunately I fear that this hasn't been the last time either.

And it was also good to look after things again in `ip link add` and friends as I realized that `0` is a valid value for txqueuelen which I previously ruled out by mistake. Thus, I started again and created a customized config parser where I chose `UINT32_MAX` to take the role of the magic value representing the directive being unset while on the other hand disallowing it as a valid input value as explained in the modified commit message. So please take another look - I am looking forward for feedback.